### PR TITLE
Add automated tests for data transforms labs

### PR DESCRIPTION
**Pull request by JakeSCahill**
Uses `doc-detective-core` to add tests for all data transforms labs published on docs.redpanda.com.

### CODE REVIEW

File: data-transforms/flatten/README.adoc [hunk 1]
```diff
@@ -94,27 +99,36 @@ export REDPANDA_CONSOLE_VERSION={latest-console-version}
...
 ```
+// (step {"action":"runShell", "command": "docker compose up -d --wait"})
+// Wait for the containers to finish getting up and running
+// (step {"action":"wait", "duration": 10000})
 
 . Set up your rpk profile:
 +
 ```bash
-rpk profile create flatten --from-profile profile.yml
+rpk profile create flatten
```
Comment (rockwotj) [inline data-transforms/flatten/README.adoc:111]: Why did we remove the --from-profile?

File: data-transforms/jq/README.adoc [hunk 2]
```diff
@@ -71,25 +76,35 @@ export REDPANDA_CONSOLE_VERSION={latest-console-version}
 ```bash
 docker compose up -d
```
Comment (rockwotj) [inline data-transforms/jq/README.adoc:77]: Did we want to add `--wait` here too?

**Review by rockwotj — APPROVED** [review]
A couple of small things. I'd love to lower the friction of writing these so that the default mode just ran all the codeblocks, but I'm not sure how large of a lift that would be.

File: data-transforms/jq/README.adoc (hunk 2 above)
Comment (JakeSCahill) [inline data-transforms/jq/README.adoc:77]:
```suggestion
docker compose up -d --wait
```

File: data-transforms/flatten/README.adoc (hunk 1 above)
Comment (JakeSCahill) [inline data-transforms/flatten/README.adoc:111]:
```suggestion
rpk profile create flatten --from-profile profile.yml
```

**Comment by JakeSCahill** [comment]
> A couple of small things. I'd love to lower the friction of writing these so that the default mode just ran all the codeblocks, but I'm not sure how large of a lift that would be.

Thanks @rockwotj - I applied your suggestions.

Regarding lowering friction, Doc Detective is a fairly new OS project. We're actively contributing to it with feature requests and bug fixes.  It's currently possible to get the tool to detect steps to run, but it's more reliable to do it this way for now.

**Review by rockwotj — APPROVED** [review]
Understood on the tool being young, thanks for doing this!

**Pull request merged**